privacyidea/lib/utils.py

```diff
@@ -250,21 +250,19 @@ def generate_password(size=6, characters=string.ascii_lowercase +
 
 def modhex_encode(s):
     return ''.join(
-        [hex2ModDict[c] for c in s.encode('hex')]
+        [hex2ModDict[c] for c in binascii.hexlify(s)]
```

Comment:
I guess this won't work on Python 3 because ``binascii.hexlify(s)`` is ``bytes``, so all ``c`` will be ``int``, but ``hex2ModDict`` has ``str`` keys :-/
... I guess we would need to change ``hexHexChars`` to a bytestring? But then we also need to change ``modhex_decode``

Reply:
Nice fix! There may be a remaining issue with the decoding function: I *think* ``modhex_decode`` will fail on Python 3 if its arguments is bytes. I think it makes sense to expect unicode strings only, but from reading the code, I'm not sure what type the arguments will actually have :-/ 
... maybe we'll just keep it like this for now, and test the yubikey token later on Python 3?

Reply:
:+1: